### PR TITLE
CLOUP-187016: Improve e2e test coverage for atlas setup command

### PR DESCRIPTION
**Pull request by VincentPikand**
## Proposed changes

Failure flow tests that were added:
- invalid projectID test
- invalid public key test
- invalid private key test

Flags that are now tested:
- `--accessListIp`
- `--projectId`

A full description of changes can be found in the JIRA ticket.

_Jira ticket:_ CLOUDP-187016

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

Not sure if I should supplement the `test/README.md` since none of the other commands have variations with flags and/or environment variables.

### CODE REVIEW

File: test/e2e/atlas/setup_force_test.go [hunk 1]
```diff
@@ -56,13 +56,40 @@ func TestSetup(t *testing.T) {
...
+
+		found := false
+		for i := range entries.Results {
+			if entries.Results[i].GetIpAddress() == randomAccessListIP {
+				found = true
+				break
+			}
+		}
+
+		req.True(found)
```
Comment (tibulca) [inline test/e2e/atlas/setup_force_test.go:90]:
```suggestion
		req.True(found, randomAccessListIP + " not found in project's access IP list")
```

**Review by tibulca — DISMISSED** [review]
LGTM, only one small suggestion

File: test/e2e/atlas/setup_failure_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 MongoDB Inc
```
Comment (andreaangiolillo) [inline test/e2e/atlas/setup_failure_test.go:1]:
```suggestion
// Copyright 2023 MongoDB Inc
```

File: test/e2e/atlas/setup_failure_test.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+func TestSetupFailureFlow(t *testing.T) {
+	g := newAtlasE2ETestGenerator(t)
+	g.generateProject("setup")
+	cliPath, err := e2e.AtlasCLIBin()
+	req := require.New(t)
+	req.NoError(err)
+
+	t.Run("Invalid Public Key", func(t *testing.T) {
+		t.Setenv("MCLI_PUBLIC_API_KEY", "invalid_public_key")
```
Comment (andreaangiolillo) [inline test/e2e/atlas/setup_failure_test.go:37]: I am wondering if changing this env would have an effect also on other tests 🤔 Could we store the original value with `GetEnv` and then use again `SetEnv` at the end of the test to set the original value?

File: test/e2e/atlas/setup_failure_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		req.Error(err)
+		assert.Contains(t, string(resp), "Unauthorized", "Expected unauthorized error due to invalid public key.")
+	})
+
+	t.Run("Invalid Private Key", func(t *testing.T) {
+		t.Setenv("MCLI_PRIVATE_API_KEY", "invalid_private_key")
+		cmd := exec.Command(cliPath,
+			"setup",
```
Comment (andreaangiolillo) [inline test/e2e/atlas/setup_failure_test.go:52]: Can you add this as a constant in [here](https://github.com/mongodb/mongodb-atlas-cli/blob/09a1900f0c02caa2df2a8321bec4eb413393786a/test/e2e/atlas/helper_test.go#L37)?

File: test/e2e/atlas/setup_failure_test.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+			"--skipSampleData",
+			"--force")
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		req.Error(err)
+		assert.Contains(t, string(resp), "Unauthorized", "Expected unauthorized error due to invalid public key.")
+	})
+
+	t.Run("Invalid Private Key", func(t *testing.T) {
+		t.Setenv("MCLI_PRIVATE_API_KEY", "invalid_private_key")
```
Comment (andreaangiolillo) [inline test/e2e/atlas/setup_failure_test.go:50]: same as above

File: test/e2e/atlas/setup_force_test.go (hunk 1 above)
Comment (gssbzn) [inline test/e2e/atlas/setup_force_test.go:90]:
```suggestion
		req.Contains(entries.Results[i].GetIpAddress(), randomAccessListIP)
```
https://pkg.go.dev/github.com/stretchr/testify@v1.8.4/assert#Contains

File: test/e2e/atlas/setup_failure_test.go (hunk 3 above)
Comment (VincentPikand) [reply test/e2e/atlas/setup_failure_test.go:37]: Here is your answer:

```go
func (*testing.T).Setenv(key string, value string)
Setenv calls os.Setenv(key, value) and uses Cleanup to restore the environment variable to its original value after the test.
```

Comment (gssbzn) [reply test/e2e/atlas/setup_failure_test.go:37]: https://pkg.go.dev/testing#T.Setenv

>Setenv calls os.Setenv(key, value) and uses Cleanup to restore the environment variable to its original value after the test.

File: test/e2e/atlas/setup_force_test.go [hunk 6]
```diff
@@ -46,6 +45,7 @@ func TestSetup(t *testing.T) {
 
 	tagKey := "env"
 	tagValue := "e2etest"
+	randomAccessListIP := "21.150.105.221"
```
Comment (gssbzn) [inline test/e2e/atlas/setup_force_test.go:48]: this doesn't seem random see
https://github.com/mongodb/mongodb-atlas-cli/blob/9152cc7651799ad9b3bd1b106046f39252ebe0ab/test/e2e/atlas/access_lists_test.go#L36-L41

File: test/e2e/atlas/setup_failure_test.go (hunk 3 above)
Comment (andreaangiolillo) [reply test/e2e/atlas/setup_failure_test.go:37]: Thanks!

File: test/e2e/atlas/setup_force_test.go (hunk 6 above)
Comment (VincentPikand) [reply test/e2e/atlas/setup_force_test.go:48]: https://en.wikipedia.org/wiki/KISS_principle

Comment (gssbzn) [reply test/e2e/atlas/setup_force_test.go:48]: still not random

Comment (VincentPikand) [reply test/e2e/atlas/setup_force_test.go]: changed it to `arbitraryAccessListIP`

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Review by gssbzn — APPROVED** [review]

**Pull request merged**